src/AbstractUpdateAction.php

```diff
@@ -65,66 +65,115 @@ public function getIndex()
     }
 
     /**
-     * Sets the version of a document for use with optimistic concurrency control.
+     * Sets the version parameters of a document for use with optimistic concurrency control.
      *
-     * @param int $version Document version
+     * @return $this
+     */
+    public function setVersionParams(array $responseData): self
```

Comment:
Do we still need this method or could we remove it?

Reply:
its still used to set the document version on GET